### DISCUSSION TITLE
No refresh when redirecting to the current url

### DISCUSSION THREAD

**Discussion by oran1248** [vercel/next.js#12939] [Help]
Hi,
The expected behavior when clicking on link that points to the current url, is a page refresh.
But when I do `Router.push(url)` and url is the current url, nothing happends.
(Maybe related to dev mode?)
Couldn't find the answer in the docs.
Thanks

**Comment by rajpraful** [comment]
You can try Router.reload() to reload the current page.

**Reply by oran1248** [reply]
But I have a link button, so I need to check the current url? something like this:
`
if(currentUrl === "X") { Router.reload() } else { Router.push("X") }
`
?

**Reply by rajpraful** [reply]
yes, you can do that. 

You will get the current URL inside router object only as router.asPath.

**Reply by oran1248** [reply]
What do you mean? can you elaborate?
And I would expect for something like: `Router.push(url, { hard: true })` or something like that, but I can also add an if statement.

**Reply by rajpraful** [reply]
try this.

`router.asPath == 'X' ? Router.reload() : Router.push('X')`

**Reply by oran1248** [reply]
what is `router`? Can i get the current url from `Router`?

**Reply by rajpraful** [reply]
Yes, Router.router.asPath

**Reply by oran1248** [reply]
Thanks it worked. there is no way to reload without loading the whole page? Like when you perform Router.push and it's different than `window.location=`

**Reply by rajpraful** [reply]
Try Router.reload('/') in place of Router.reload()

**Reply by oran1248** [reply]
It doesn't seem to work :(
The same as Router.reload()

**Comment by timneutkens** [comment]
Opened a RFC for redirecting here: #14890
